### PR TITLE
Removed deprecated evented api method

### LINKED ISSUE #4024: Remove deprecation `ember-data:evented-api-usage`
Due to the following depreciation, it is very hard to read the travis logs. For every single test there are 10-15 depreciation of id `ember-data:evented-api-usage`

```sh
 WARN: DEPRECATION: Called ready on event#39ae7368 [deprecation id: ember-data:evented-api-usage] See https://deprecations.emberjs.com/ember-data/v3.x/#deprecatingrecordlifecycleeventmethods for more details.
3950                    at logDeprecationStackTrace (http://localhost:7357/assets/vendor.js:59858:21)
3951                    at HANDLERS.<computed> (http://localhost:7357/assets/vendor.js:59956:9)
3952                    at raiseOnDeprecation (http://localhost:7357/assets/vendor.js:59885:9)
3953                    at HANDLERS.<computed> (http://localhost:7357/assets/vendor.js:59956:9)
3954                    at handleDeprecationWorkflow (http://localhost:7357/assets/vendor.js:88577:9)
3955                    at HANDLERS.<computed> (http://localhost:7357/assets/vendor.js:59956:9)
3956                    at deprecationCollector (http://localhost:7357/assets/vendor.js:88602:5)
3957                    at HANDLERS.<computed> (http://localhost:7357/assets/vendor.js:59956:9)
3958                    at invoke (http://localhost:7357/assets/vendor.js:59968:9)
```

### PR DESCRIPTION
**Pull request by progmatic-99**
Fixes #4024 

#### Short description of what this resolves:
Removed `ready()` method under evented API that was going to be deprecated with ember 4.0

### CODE REVIEW

File: app/models/attendee.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default ModelBase.extend({
   ticket : belongsTo('ticket'),
   user   : belongsTo('user'),
 
-  ready() {
+  isReady() {
```
Comment (iamareebjamal) [inline app/models/attendee.js:47]: Is it actually an event?

**Comment by iamareebjamal** [comment]
Is there any docs in isReady calback. I thought ready had no alternative

**Comment by progmatic-99** [comment]
Yes, `ready` has no alternative. I am changing this.

**Comment by iamareebjamal** [comment]
Then why did you even create a PR with it, I'm confused?

**Comment by progmatic-99** [comment]
I wanted to see the travis logs when i removed the event. 
Should i create a custom event that behaves like `ready` or should i remove `ready`???

**Comment by iamareebjamal** [comment]
If you remove ready, what will happen to the code that's supposed to run in that callback

File: app/models/attendee.js [hunk 2]
```diff
@@ -34,19 +34,13 @@ export default ModelBase.extend({
   gender             : attr('string'),
   ageGroup           : attr('string'),
   birthDate          : attr('moment'),
-  complexFieldValues : attr(),
+  complexFieldValues : attr({ defaultValue: () => {} }),
```
Comment (iamareebjamal) [inline app/models/attendee.js:37]: That's default value set to an empty function

File: app/models/custom-form.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ export default ModelBase.extend({
   fieldIdentifier : attr('string'),
   form            : attr('string'),
   type            : attr('string', { defaultValue: 'text' }),
-  name            : attr('string'),
+  _name           : attr('string'),
```
Comment (iamareebjamal) [inline app/models/custom-form.js:54]: This won't work

File: app/models/attendee.js (hunk 2 above)
Comment (progmatic-99) [reply app/models/attendee.js:37]: but it will return an empty object

File: app/models/custom-form.js (hunk 3 above)
Comment (progmatic-99) [reply app/models/custom-form.js:54]: i have defined a computed property `name` which will use `_name`

Comment (iamareebjamal) [reply app/models/custom-form.js:54]: Won't work. Try the custom forms and attach a GIF

File: app/models/attendee.js (hunk 2 above)
Comment (iamareebjamal) [reply app/models/attendee.js:37]: No

File: app/models/custom-form.js (hunk 3 above)
Comment (progmatic-99) [reply app/models/custom-form.js:54]: are you referring to the forms  under session & speaker details?

Comment (iamareebjamal) [reply app/models/custom-form.js:54]: Yes

**Comment by iamareebjamal** [comment]
Please reopen when you have a solution

**Pull request closed without merging**